From b7e7d91abbe8cc324bac8f0f6fbf2a67b459b94b Mon Sep 17 00:00:00 2001
From: Chen Minqiang <ptpt52@gmail.com>
Date: Wed, 21 Apr 2021 13:27:42 +0800
Subject: [PATCH 2/5] kernel: add kmod-ipvlan support

This kmod is similar to macvlan with the difference being that the
endpoints have the same mac address.

It is useful on cloud where only one mac address allowed on port,
where macvlan not works but ipvlan would.

One use case is where multiple IPs and gateways assign on one net port

Signed-off-by: Chen Minqiang <ptpt52@gmail.com>
---
 package/kernel/linux/modules/netdevices.mk | 16 ++++++++++++++++
 target/linux/generic/config-5.4            |  1 +
 2 files changed, 17 insertions(+)

diff --git a/package/kernel/linux/modules/netdevices.mk b/package/kernel/linux/modules/netdevices.mk
index 901bddddea..f75c1b9ea1 100644
--- a/package/kernel/linux/modules/netdevices.mk
+++ b/package/kernel/linux/modules/netdevices.mk
@@ -865,6 +865,22 @@ endef
 $(eval $(call KernelPackage,macvlan))
 
 
+define KernelPackage/ipvlan
+  SUBMENU:=$(NETWORK_DEVICES_MENU)
+  TITLE:=IP-VLAN support
+  KCONFIG:=CONFIG_IPVLAN
+  FILES:=$(LINUX_DIR)/drivers/net/ipvlan/ipvlan.ko
+  AUTOLOAD:=$(call AutoProbe,ipvlan)
+endef
+
+define KernelPackage/ipvlan/description
+ A kernel module which allows one to create virtual interfaces that
+ map packets to or from specific IP addresses to a particular interface
+endef
+
+$(eval $(call KernelPackage,ipvlan))
+
+
 define KernelPackage/tulip
   TITLE:=Tulip family network device support
   DEPENDS:=@PCI_SUPPORT +kmod-mii
diff --git a/target/linux/generic/config-5.4 b/target/linux/generic/config-5.4
index be025d2073..5ba7401673 100644
--- a/target/linux/generic/config-5.4
+++ b/target/linux/generic/config-5.4
@@ -2448,6 +2448,7 @@ CONFIG_IPV6_NDISC_NODETYPE=y
 # CONFIG_IPV6_TUNNEL is not set
 # CONFIG_IPV6_VTI is not set
 # CONFIG_IPVLAN is not set
+# CONFIG_IPVTAP is not set
 # CONFIG_IPW2100 is not set
 # CONFIG_IPW2100_DEBUG is not set
 CONFIG_IPW2100_MONITOR=y
-- 
2.34.1

